plugin/trino-redshift/src/test/java/io/trino/plugin/redshift/TestRedshiftUnload.java

```diff
@@ -0,0 +1,239 @@
+import static io.trino.plugin.redshift.RedshiftQueryRunner.AWS_SECRET_KEY;
+import static io.trino.plugin.redshift.RedshiftQueryRunner.IAM_ROLE;
+import static io.trino.plugin.redshift.RedshiftQueryRunner.S3_UNLOAD_ROOT;
+import static io.trino.plugin.redshift.TestingRedshiftServer.JDBC_PASSWORD;
+import static io.trino.plugin.redshift.TestingRedshiftServer.JDBC_URL;
+import static io.trino.plugin.redshift.TestingRedshiftServer.JDBC_USER;
+import static io.trino.plugin.redshift.TestingRedshiftServer.TEST_DATABASE;
+import static io.trino.plugin.redshift.TestingRedshiftServer.TEST_SCHEMA;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.TestInstance.Lifecycle.PER_CLASS;
+import static org.junit.jupiter.api.parallel.ExecutionMode.CONCURRENT;
+
+@TestInstance(PER_CLASS)
+@Execution(CONCURRENT)
+public class TestRedshiftUnload
+        extends AbstractTestQueries
+{
+    private RedshiftConfig redshiftUnloadConfig;
+    private RedshiftSplitManager redshiftSplitManager;
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        redshiftUnloadConfig = new RedshiftConfig().setUnloadLocation(S3_UNLOAD_ROOT).setUnloadIamRole(IAM_ROLE);
+        redshiftSplitManager = createSplitManager();
+        return RedshiftQueryRunner.builder()
+                // NOTE this can cause tests to time-out if larger tables like
+                //  lineitem and orders need to be re-created.
+                .setInitialTables(REQUIRED_TPCH_TABLES)
+                .build();
+    }
+
+    private RedshiftSplitManager createSplitManager()
+    {
+        DriverConnectionFactory driverConnectionFactory = DriverConnectionFactory.builder(new Driver(), JDBC_URL, new ExtraCredentialProvider(Optional.empty(), Optional.empty(), new StaticCredentialProvider(Optional.of(JDBC_USER), Optional.of(JDBC_PASSWORD))))
+                .build();
+        RedshiftClient redshiftClient = new RedshiftClient(
+                new BaseJdbcConfig().setConnectionUrl(JDBC_URL),
+                redshiftUnloadConfig,
+                driverConnectionFactory,
+                new JdbcStatisticsConfig(),
+                new DefaultQueryBuilder(NONE),
+                new DefaultIdentifierMapping(),
+                NONE);
+
+        return new RedshiftSplitManager(
+                redshiftClient,
+                new DefaultQueryBuilder(NONE),
+                NONE,
+                new JdbcSplitManager(redshiftClient),
+                redshiftUnloadConfig,
+                new S3FileSystemFactory(OpenTelemetry.noop(), new S3FileSystemConfig()
+                        .setAwsAccessKey(AWS_ACCESS_KEY)
+                        .setAwsSecretKey(AWS_SECRET_KEY)
+                        .setRegion(AWS_REGION)
+                        .setStreamingPartSize(DataSize.valueOf("5.5MB")), new S3FileSystemStats()),
+                Executors.newFixedThreadPool(1));
+    }
+
+    @Test
+    void testUnloadFlow()
```

Comment:
You can get splits count by looking at driver count and the paths from looking at operator split info from a normal query execution as well. It's not necessary to hand craft this test.
See usages of `io.trino.testing.AbstractTestQueryFramework#assertQueryStats`

Reply:
This test explicitly verifies that `RedshiftUnloadSplitSource` was kicked in at least for some kind of query as we have a fall back to JDBC in case of unload query building failure https://github.com/trinodb/trino/pull/24117/files#diff-7ac016bcb60087e0bd185a0f809bd98c59df9064cc3a27aaaf2c362dd682023bR128-R134
Incase if unload query builder always fails then, all of the queries will silently fall back to JDBC.

Reply:
Did you try looking at operator stats of the scan ? I think `io.trino.operator.OperatorStats#info` should contain `io.trino.plugin.redshift.RedshiftUnloadSplit#getSplitInfo` which should allow detecting whether UNLOAD was used. If that doesn't work, you should also see parquet reader metrics in `connectorMetrics` through `io.trino.plugin.redshift.RedshiftParquetPageSource#getMetrics`.
There have to be indications in the query metrics about UNLOAD for our own debugging and telemetry as well.

Reply:
thanks for providing further pointers on it. I changed the test to use assert using info from `io.trino.plugin.redshift.RedshiftUnloadSplit#getSplitInfo`. 
JDBC flow doesn't produce OperatorInfo as asserted in `testUnloadFlowFallbackToJdbc`